frb_codegen/src/generator/dart/mod.rs

```diff
@@ -221,48 +261,156 @@ fn generate_dart_declaration_body(
+    let DartApiSpec {
+        dart_funcs,
+        dart_api2wire_funcs,
+        dart_api_fill_to_wire_funcs,
+        dart_wire2api_funcs,
+        dart_wasm_funcs,
+        dart_wasm_module,
+        dart_structs: _,
+        needs_freezed: _,
+    } = spec;
+
+    lines.push_acc(Acc::new(|target| match target {
+        Common => format!(
+            "class {impl} implements {} {{
+                final {plat} _plat;
+                factory {impl}(ExternalLibrary dylib) => {impl}.raw({plat}(dylib));
+
+                /// Only valid on web/WASM platforms.
+                factory {impl}.wasm(FutureOr<WasmModule> module) =>
+                    {impl}(module as ExternalLibrary);
+                {impl}.raw(this._plat);",
+            dart_api_class_name,
+            impl = dart_api_impl_class_name,
+            plat = dart_platform_class_name,
+        ),
+        Io => format!(
+            "class {plat} extends FlutterRustBridgeBase<{wire}> {{
+                {plat}(ffi.DynamicLibrary dylib) : super({wire}(dylib));",
+            plat = dart_platform_class_name,
+            wire = dart_wire_class_name,
+        ),
+        Wasm => format!(
+            "class {plat} extends FlutterRustBridgeBase<{wire}> with FlutterRustBridgeSetupMixin {{
+                {plat}(FutureOr<WasmModule> dylib) : super({wire}(dylib)) {{
+                    setupMixinConstructor();
+                }}
+                Future<void> setup() => inner.init;",
+            plat = dart_platform_class_name,
+            wire = dart_wire_class_name,
+        ),
+    }));
 
-            // Section: api_fill_to_wire
-            {}
-        }}
+    lines.extend(dart_funcs.iter().map(|func| {
+        format!(
+            "{}\n\n{}",
+            func.implementation, func.companion_field_implementation,
+        )
+    }));
+
+    lines.push("}\n".into());
+
+    lines.push_all(section_header("api2wire"));
+    lines.push_acc(dart_api2wire_funcs.clone());
+
+    lines.io.push(section_header("api_fill_to_wire"));
+    lines.io.push(dart_api_fill_to_wire_funcs.join("\n\n"));
+
+    lines.push_acc(Acc::new(|target| match target {
+        Io | Wasm => "}\n".into(),
+        Common => "".into(),
+    }));
+
+    lines.push(section_header("wire2api"));
+    lines.push(dart_wire2api_funcs.join("\n\n"));
+
+    if config.wasm_enabled {
+        let dart_wasm_module_name = config.dart_wasm_module();
+        lines.wasm.push(section_header("WASM wire module"));
+        lines.wasm.push(format!(
+            "@JS('wasm_bindgen') external {} get wasmModule;",
+            dart_wasm_module_name,
+        ));
+        lines.wasm.push(format!(
+            "@JS() @anonymous class {wasm} implements WasmModule {{
+                external Object /* Promise */ call([String? moduleName]);
+                external {wasm} bind(dynamic thisArg, String moduleName);
+            ",
+            wasm = dart_wasm_module_name,
+        ));
```

Comment:
nit: is it possible we extract it to somewhere else, or at least a function, since we are in `mod.rs`

Btw would be great if the `mod.rs` could be refactored a bit since some code looks like they do not need to be in `mod.rs` and can be moved to new files etc.